EpsilonAuxBMS/Src/SetContactorsTask.c

```diff
@@ -138,90 +158,127 @@ void setContactorsTask(void const* arg)
 
                 if (!contactorError)
                 {
-                    contactorState = ON;
                     // Enable high voltage
                     HAL_GPIO_WritePin(HV_ENABLE_GPIO_Port, HV_ENABLE_Pin, GPIO_PIN_SET);
+                    state = DONE;
                 }
-                else
+
+                if (osMutexWait(auxStatus.auxStatusMutex, 0) != osOK)
                 {
-                    contactorState = OFF;
+                    continue;
                 }
 
-                if (updateContactorState(contactorState, contactorError, DISCHARGE))
+                auxStatus.contactorError = contactorError;
+                auxStatus.startUpSequenceDone = 1;
+
+                osMutexRelease(auxStatus.auxStatusMutex);
+
+
+                common = !HAL_GPIO_ReadPin(COMMON_SENSE_GPIO_Port, COMMON_SENSE_Pin);
+                charge = !HAL_GPIO_ReadPin(CHARGE_SENSE_GPIO_Port, CHARGE_SENSE_Pin);
+
+                if (!charge || !common) // Charge or common contactor have been disconnected
                 {
-                    state = DONE;
+                    disconnectContactors(1);
+                    state = CONTACTOR_DISCONNECTED;
                 }
 
                 break;
 
             case DONE:
             {
-                uint8_t common = !HAL_GPIO_ReadPin(COMMON_SENSE_GPIO_Port, COMMON_SENSE_Pin);
-                uint8_t charge = !HAL_GPIO_ReadPin(CHARGE_SENSE_GPIO_Port, CHARGE_SENSE_Pin);
-                uint8_t discharge = !HAL_GPIO_ReadPin(DISCHARGE_SENSE_GPIO_Port, DISCHARGE_SENSE_Pin);
+                common = !HAL_GPIO_ReadPin(COMMON_SENSE_GPIO_Port, COMMON_SENSE_Pin);
+                charge = !HAL_GPIO_ReadPin(CHARGE_SENSE_GPIO_Port, CHARGE_SENSE_Pin);
+                discharge = !HAL_GPIO_ReadPin(DISCHARGE_SENSE_GPIO_Port, DISCHARGE_SENSE_Pin);
 
-                if (!common || !charge || !discharge)
+                uint8_t turningOnCharge = 0;
```

Comment:
I can't fully remember, but this variable seems to be for displaying if the charge contactor is currently in the process of being turned on. Maybe a better name would be 'isChargeTurningOn'